tests/ui/components/search/searchcomponent.js

```diff
@@ -132,15 +135,71 @@ describe('SearchBar component', () => {
     expect(storage.get(StorageKeys.QUERY_TRIGGER)).toEqual(QueryTriggers.SEARCH_BAR);
   });
 
-  it('submitVoiceQuery sets QUERY_TRIGGER to VOICE_SEARCH', () => {
-    const component = COMPONENT_MANAGER.create('SearchBar', {
-      ...defaultConfig,
-      voiceSearch: {
-        enabled: true
-      }
+  describe('Voice search works properly', () => {
+    const events = {};
+    const webkitSpeechRecognitionMock = {
+      addEventListener: jest.fn((eventName, cb) => {
+        events[eventName] = cb;
+      }),
+      start: jest.fn(() => {
+        'start' in events && events.start();
+      }),
+      stop: jest.fn(() => {
+        'stop' in events && events.stop();
+      })
+    };
+
+    beforeEach(() => {
+      jest.spyOn(global, 'window', 'get').mockImplementation(() => ({
```

Comment:
is the purpose of jest.spyOn purely for setting a mockImplementation? and if so, why not just jest.fn()?

Reply:
I tried putting a mock function on the window and I wasn't able to get it working, however I was able to get the spy working so I went with that